src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,95 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int START_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[START_CAPACITY];
+        threshold = (int) (START_CAPACITY * LOAD_FACTOR);
+    }
+
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        if (table[index] != null) {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (current.key == key || current.key != null && current.key.equals(key)) {
+                    current.value = value;
+                    break;
+                }
+                if (current.next == null) {
+                    current.next = new Node<>(null, value, key, hash(key));
+                    size++;
+                    break;
+                }
+                current = current.next;
+            }
+        } else {
+            table[index] = new Node<>(null, value, key, hash(key));
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (table[index] != null) {
+            Node<K, V> node = table[index];
+            while (node != null) {
+                if (node.key == key || node.key != null && node.key.equals(key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private class Node<K,V> {
+        private Node<K, V> next;
+        private V value;
+        private K key;
+        private int hash;
```

Comment:
Why do you need this field?

Reply:
I just implement class Node like in original Java HashMap Node class